src/mgr/StandbyPyModules.cc

```diff
@@ -72,29 +75,30 @@ void StandbyPyModules::shutdown()
   modules.clear();
 }
 
-int StandbyPyModules::start_one(PyModuleRef py_module)
+void StandbyPyModules::start_one(PyModuleRef py_module)
 {
   std::lock_guard l(lock);
-  const std::string &module_name = py_module->get_name();
-
-  ceph_assert(modules.count(module_name) == 0);
-
-  modules[module_name].reset(new StandbyPyModule(
-      state,
-      py_module, clog));
-
-  int r = modules[module_name]->load();
-  if (r != 0) {
-    modules.erase(module_name);
-    return r;
-  } else {
-    dout(4) << "Starting thread for " << module_name << dendl;
-    // Giving Thread the module's module_name member as its
-    // char* thread name: thread must not outlive module class lifetime.
-    modules[module_name]->thread.create(
-        modules[module_name]->get_name().c_str());
-    return 0;
-  }
+  auto name = py_module->get_name();
+
+  ceph_assert(modules.count(name) == 0);
+
+  modules[name].reset(new StandbyPyModule(state, py_module, clog));
+  auto standby_module = modules.at(name).get();
+
+  // Send all python calls down a Finisher to avoid blocking
+  // C++ code, and avoid any potential lock cycles.
+  finisher.queue(new FunctionContext([this, standby_module, name](int) {
+    std::lock_guard l(lock);
```

Comment:
why do you think it's different? we still hold the `lock` when calling `standby_module->load()` after this change.

Reply:
ic. and agreed. we don't need to hold the lock unless we are going to access `this->modules`.

Reply:
done